pkg/xtrabackup/sidecarclient.go

```diff
@@ -0,0 +1,96 @@
+package xtrabackup
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/percona/percona-server-mysql-operator/pkg/mysql"
+	"github.com/pkg/errors"
+)
+
+type SidecarClient interface {
+	GetRunningBackupConfig(ctx context.Context) (*BackupConfig, error)
+	DeleteBackup(ctx context.Context, name string, cfg BackupConfig) error
```

Comment:
let's rename this to `DeleteBackupConfig`. its name is misleading now, it makes you think it'll delete the backup files